File: ResourceManager/Services/Abstractions/IRoleDataService.cs
using ResourceManager.Models;
using System.Collections.Generic;
using System.Threading.Tasks;

namespace ResourceManager.Services.Abstractions
{
    public interface IRoleDataService : IDataService<Role>
    {
        Task<IList<string>> GetRoleStringForUserID(int iD);
        Task<IList<Role>> GetRoleObjectForUserID(int iD);
        Task<int> UpdatePermissions(Role role, IList<Permission> permissions);
    }
}

File: ResourceManager/Services/Abstractions/IRoleDataService.vb
Imports ResourceManager.Models
Imports System.Collections.Generic
Imports System.Threading.Tasks

Namespace ResourceManager.Services.Abstractions
    Public Interface IRoleDataService
        Inherits IDataService(Of Role)
        Function GetRoleStringForUserID(ByVal iD As Integer) As Task(Of IList(Of String))
        Function GetRoleObjectForUserID(ByVal iD As Integer) As Task(Of IList(Of Role))
        Function UpdatePermissions(ByVal role As Role, ByVal permissions As IList(Of Permission)) As Task(Of Integer)
    End Interface
End Namespace
